src/directives/NgGrid.ts

```diff
@@ -786,13 +792,16 @@ export class NgGrid implements OnInit, DoCheck, OnDestroy {
 			}
 			
 			if (shouldSave) {
-				collisions[0].savePosition(itemPos);
+				const newItemPos = this._fixGridPosition(itemPos, itemDims);
+				collisions[0].savePosition(newItemPos.col, newItemPos.row);
 			} else {
-				collisions[0].setGridPosition(itemPos);
+		                const newItemPos = this._fixGridPosition(itemPos, itemDims);

```

Comment:
I actually removed that line in my own setup, but I imagined there was a reason this was in the code already. I should have commented that in my commit, which I forgot. I'm sorry for the mistake on my part.